### PR TITLE
Cleanup identity-providers feature (Pt.09)

### PR DESCRIPTION
**Pull request by JayaShakthi97**
### Purpose
- Clean up constants in `admin.connections` feature.
- Remove redundant constant files from `admin.identity-providers` feature.
- Refactor connections' constants structure.

### Related Issues
- https://github.com/wso2/product-is/issues/20580

### Related PRs
- N/A

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: features/admin.applications.v1/components/settings/sign-on-methods/step-based-flow/add-authenticator-modal.tsx [hunk 1]
```diff
@@ -23,6 +23,9 @@ import {
     ConnectionTemplateInterface
 } from "@wso2is/admin.connections.v1";
 import { useGetConnectionTemplates } from "@wso2is/admin.connections.v1/api/use-get-connection-templates";
+import {
+    CommonAuthenticatorManagementConstants
+} from "@wso2is/admin.connections.v1/constants/common-authenticator-constants";
```
Comment (pavinduLakshan) [inline features/admin.applications.v1/components/settings/sign-on-methods/step-based-flow/add-authenticator-modal.tsx:28]:
```suggestion
import { CommonAuthenticatorManagementConstants } from "@wso2is/admin.connections.v1/constants/common-authenticator-constants";
```

File: features/admin.applications.v1/constants/application-management.ts [hunk 2]
```diff
@@ -17,6 +17,10 @@
  */
 
 import { ConnectionManagementConstants } from "@wso2is/admin.connections.v1/constants/connection-constants";
+import {
+    FederatedAuthenticatorConstants
+} from "@wso2is/admin.connections.v1/constants/federated-authenticator-constants";
```
Comment (pavinduLakshan) [inline features/admin.applications.v1/constants/application-management.ts:22]:
```suggestion
import { FederatedAuthenticatorConstants } from "@wso2is/admin.connections.v1/constants/federated-authenticator-constants";
```

File: features/admin.applications.v1/pages/application-edit.tsx [hunk 3]
```diff
@@ -87,12 +88,14 @@ const ApplicationEditPage: FunctionComponent<ApplicationEditPageInterface> = (
...
     const appDescElement: React.MutableRefObject<HTMLDivElement> = useRef<HTMLDivElement>(null);
 
-    const allowedScopes: string = useSelector((state: AppState) => state?.auth?.allowedScopes);
     const applicationTemplates: ApplicationTemplateListItemInterface[] = useSelector(
         (state: AppState) => state.application.templates);
     const featureConfig: FeatureConfigInterface = useSelector((state: AppState) => state.config.ui.features);
     const tenantDomain: string = useSelector((state: AppState) => state.auth.tenantDomain);
 
+    // Check if the user has the required scopes to update the application.
+    const hasUsersUpdatePermissions: boolean = useRequiredScopes(featureConfig?.applications?.scopes?.update);
```
Comment (pavinduLakshan) [inline features/admin.applications.v1/pages/application-edit.tsx:97]: Convention has been `has<feature><action>Permissions`. Let's stick to that.

```suggestion
    const hasApplicationUpdatePermissions: boolean = useRequiredScopes(featureConfig?.applications?.scopes?.update);
```

File: features/admin.connections.v1/components/edit/connection-edit.tsx [hunk 4]
```diff
@@ -171,10 +168,7 @@ export const EditConnection: FunctionComponent<EditConnectionPropsInterface> = (
     const [ isTrustedTokenIssuer, setIsTrustedTokenIssuer ] = useState<boolean>(false);
     const [ isExpertMode, setIsExpertMode ] = useState<boolean>(false);
 
-    const isApplicationReadAccessAllowed: boolean = useMemo(() => (
-        hasRequiredScopes(
-            featureConfig?.applications, featureConfig?.applications?.scopes?.read, allowedScopes)
-    ), [ featureConfig, allowedScopes ]);
+    const isApplicationReadAccessAllowed: boolean = useRequiredScopes(featureConfig?.applications?.scopes?.read);
```
Comment (pavinduLakshan) [inline features/admin.connections.v1/components/edit/connection-edit.tsx:171]:
```suggestion
    const hasApplicationReadPermissions: boolean = useRequiredScopes(featureConfig?.applications?.scopes?.read);
```

File: features/admin.applications.v1/components/settings/sign-on-methods/step-based-flow/add-authenticator-modal.tsx (hunk 1 above)
Comment (JayaShakthi97) [reply features/admin.applications.v1/components/settings/sign-on-methods/step-based-flow/add-authenticator-modal.tsx:28]: Exceeds the maximum allowed length.

File: features/admin.applications.v1/constants/application-management.ts (hunk 2 above)
Comment (JayaShakthi97) [reply features/admin.applications.v1/constants/application-management.ts:22]: Exceeds the maximum allowed length.

File: features/admin.connections.v1/components/edit/connection-edit.tsx (hunk 4 above)
Comment (JayaShakthi97) [reply features/admin.connections.v1/components/edit/connection-edit.tsx]: Fixed with e404ed9145c6589528141f048ce9eb70cc53b864

File: features/admin.applications.v1/pages/application-edit.tsx (hunk 3 above)
Comment (JayaShakthi97) [reply features/admin.applications.v1/pages/application-edit.tsx]: Fixed with e404ed9145c6589528141f048ce9eb70cc53b864

File: features/admin.connections.v1/constants/federated-authenticator-constants.ts [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
+    /**
+     * Google One Tap enabling request parameter.
+     */
+    public static readonly GOOGLE_ONE_TAP_ENABLED_PARAM: string = "IsGoogleOneTapEnabled";
+
+    /**
+     * Google Scope mappings.
+     */
+    public static readonly GOOGLE_SCOPE_DICTIONARY: Record<string, string> = {
```
Comment (dasuni-30) [inline features/admin.connections.v1/constants/federated-authenticator-constants.ts:59]: Shall we use a single constant for these two?

**Review by dasuni-30 — APPROVED** [review]

File: features/admin.connections.v1/constants/federated-authenticator-constants.ts (hunk 5 above)
Comment (JayaShakthi97) [reply features/admin.connections.v1/constants/federated-authenticator-constants.ts:59]: Will be addressed via a followup PR

**Pull request merged**